src/object_api.c

```diff
@@ -42,6 +42,10 @@ git_repository *git_commit_owner(const git_commit *obj)
 	return git_object_owner((const git_object *)obj);
 }
 
+int git_commit_dup(git_commit **out, git_commit *obj)
+{
+	return git_object_dup((git_object **)out, (git_object *)obj);
```

Comment:
`git_object_dup()` is public, but in either case, if you're passing in the wrong type, you've purposefully done a wild cast in your code, so these would only be for convenience so the compiler sees the types it wants in the function signature.

The actual object type is irrelevant for these functions anyway, as it only touches common areas.